### PR TITLE
add parameter workflowId to avworkflow_files() and avworkflow_localize()

### PR DESCRIPTION
**Pull request by yubocheng**

### CODE REVIEW

File: R/avworkflow.R [hunk 1]
```diff
@@ -318,6 +318,9 @@ avworkflow_jobs <-
 #'     tibble with column `submissionId`, or NULL / missing. See
 #'     'Details'.
 #'
+#' @param workflowId a character() of internal identifier associated with one
+#'     workflow in the submission, or NULL / missing.
```
Comment (LiNk-NY) [inline R/avworkflow.R:322]: Please make it clear that it is either a scalar character or character vector, e.g., `character(1)`

File: R/avworkflow.R [hunk 2]
```diff
@@ -411,6 +417,18 @@ avworkflow_files <-
...
     )
 
+    if (!is.null(workflowId) && !(workflowId %in% pull(tbl, "workflowId"))) {
+        stop(.pretty_text(
+            "'workflowId' must be NULL, or character(1) associated with the ",
+            "provided submissionId"
+        ))
+    }
+
+    if (!is.null(workflowId) && workflowId %in% pull(tbl, "workflowId"))
```
Comment (LiNk-NY) [inline R/avworkflow.R]: didn't you test for the second condition already in the previous block?

File: R/avworkflow.R [hunk 3]
```diff
@@ -495,7 +515,23 @@ avworkflow_localize <-
...
+        ))
+    }
+
+    if (!is.null(workflowId) && workflowId %in% pull(fls, "workflowId"))
+        fls <- avworkflow_files(submissionId, workflowId)
+
     source <- pull(fls, "submissionRoot") |> unique()
+    if (!is.null(workflowId) && workflowId %in% pull(fls, "workflowId"))
+        source <- 
+            paste0(pull(fls, "submissionRoot") |> unique(), "/",
```
Comment (LiNk-NY) [inline R/avworkflow.R]: some of this is already the source, i.e., `source == pull(fls, "submissionRoot") |> unique()`

File: R/avworkflow.R [hunk 4]
```diff
@@ -495,7 +515,23 @@ avworkflow_localize <-
...
     fls <- avworkflow_files(submissionId)
 
+    if (!is.null(workflowId) && !(workflowId %in% pull(fls, "workflowId"))) {
+        stop(.pretty_text(
+            "'workflowId' must be NULL, or character(1) associated with the ",
+            "provided submissionId"
+        ))
+    }
+
+    if (!is.null(workflowId) && workflowId %in% pull(fls, "workflowId"))
```
Comment (LiNk-NY) [inline R/avworkflow.R]: remove redundant second condition

File: R/avworkflow.R [hunk 5]
```diff
@@ -495,7 +515,23 @@ avworkflow_localize <-
...
+        fls <- avworkflow_files(submissionId, workflowId)
+
     source <- pull(fls, "submissionRoot") |> unique()
+    if (!is.null(workflowId) && workflowId %in% pull(fls, "workflowId"))
+        source <- 
+            paste0(pull(fls, "submissionRoot") |> unique(), "/", 
+                   pull(fls, "workflow") |> unique(), "/", 
+                   workflowId)
```
Comment (LiNk-NY) [inline R/avworkflow.R]: Is there a cleaner way to do this? perhaps with `unite`?

File: R/avworkflow.R [hunk 6]
```diff
@@ -708,3 +744,75 @@ avworkflow_stop <-
...
+        .is_scalar_character(namespace),
+        .is_scalar_character(name)
+    )
+    
+   if (is.null(submissionId)) {
+        submissionId <-
+            as.character((avworkflow_jobs(namespace = namespace, name = name) |>
+                        ## default: most recent workflow job
+                        head(1))[1])
+    } else {
```
Comment (LiNk-NY) [inline R/avworkflow.R]: remove `else` here, it's not needed

File: inst/data/res1.rds
Comment (LiNk-NY) [inline inst/data/res1.rds:1]: what are these for?

File: vignettes/WorkflowInfo.Rmd [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+submissionId <- "9385fd75-4cb7-470f-9e07-1979e2c8f193"
+res1 <- avworkflow_info(submissionId)
+```
+
+Workflow information for sample submissionId 
+"9385fd75-4cb7-470f-9e07-1979e2c8f193" retrieved as above is stored 
+as res1.rds. 
+
+```{r, message=FALSE, warning=FALSE}
+res1 <- readRDS(system.file(package = "AnVIL", "data", "res1.rds"))
```
Comment (LiNk-NY) [inline vignettes/WorkflowInfo.Rmd]: Reconsider storing the results of these operations. The user should be able to run these examples on AnVIL.

**Review by LiNk-NY — CHANGES_REQUESTED** [review]
Thanks for the PR! It looks good. Please see the comments below.

**Comment by yubocheng** [comment]
Updates were made accordingly, please review. Function workflowinfo() and vignette were updated in another branch and new pull request will be created.

**Comment by LiNk-NY** [comment]
Hi Yubo, @yubocheng 
Can you respond to the points listed above or click the resolve conversation buttons?
I will take a look shortly. 
Thanks!

File: R/avworkflow.R (hunk 2 above)
Comment (yubocheng) [reply R/avworkflow.R]: Updated as suggested

File: R/avworkflow.R (hunk 3 above)
Comment (yubocheng) [reply R/avworkflow.R]: Updated as suggested

File: R/avworkflow.R (hunk 4 above)
Comment (yubocheng) [reply R/avworkflow.R]: Updated as suggested

File: R/avworkflow.R (hunk 5 above)
Comment (yubocheng) [reply R/avworkflow.R]: unite works for a tibble, so updated to 
source <- 
            paste0(source, "/", 
                   pull(fls, "workflow") |> unique(), "/", 
                   workflowId)

File: R/avworkflow.R (hunk 6 above)
Comment (yubocheng) [reply R/avworkflow.R]: Updated as suggested, and this function will be in a new pull request

File: vignettes/WorkflowInfo.Rmd (hunk 7 above)
Comment (yubocheng) [reply vignettes/WorkflowInfo.Rmd]: Will be in a new pull request. res1 and res2 are stored in /inst/data

File: inst/data/res1.rds
Comment (yubocheng) [reply inst/data/res1.rds:1]: Will be in a new pull request. res1 and res2 are results for avworkflow_info() and stored here for vignette

**Comment by yubocheng** [comment]
Hi Marcel,

I responded and clicked resolve conversation buttons. Thanks
________________________________
From: Marcel Ramos ***@***.***>
Sent: Thursday, October 19, 2023 2:17 PM
To: Bioconductor/AnVIL ***@***.***>
Cc: Cheng, Yubo ***@***.***>; Mention ***@***.***>
Subject: EXTERNAL: Re: [Bioconductor/AnVIL] add parameter workflowId to avworkflow_files() and avworkflow_localize() (PR #90)


Hi Yubo, @yubocheng<https://secure-web.cisco.com/1H8JvE_Y0a0Uxa_6oF2dX-ZyG5MH97mpLlC71Ml3DKzkfZyoqdZdLbI4pWFzPLmKqWf6GtBd1ciXrZ8ObaWPXJ71pusWYAuSiOJr1trrq8y7Kiip4I8qQ6vGgJVY86BApZrCOze5wJ_NUfmrChWC6-7KzSaKxjF012F7kUngDXBPlPRUv03GJc42RoKGwquc_n0JdFJaI3lG_H4m7NDYYHyiCRUMokfqCJcsYHRACRNp1ucg2CE_UvX_pOiJEkjTzrR42Q0-SLAQyyCRXLIPUt-4h6H0H2hZoanmk8bKY5GLA_wULFRUVrV5xEOEniuEx/https%3A%2F%2Fgithub.com%2Fyubocheng>
Can you respond to the points listed above or click the resolve conversation buttons?
I will take a look shortly.
Thanks!

—
Reply to this email directly, view it on GitHub<https://secure-web.cisco.com/1JNqHBjUhV5wbtio0LuVbAT6YkdxhrS-dQWoANn7xmiUTTcBUlKex62t7AydRIio5T8XNFMy2SSCohcVylhoiv4xPMeWnW4LHUyp3scD1YENsHvHAopi921hjSieecJA-op7VC2I6L0duEmx8aHOw2X4Ladha5v418XD0-MkmC94XGdxW1gmwMcngUZf3g-GtyKPdc2rZeLyBTSKeoUG8CaZhcUO5jEDkJPjb_Ofaiv77lpUvxphKlBTB6MCD1Rf95ptwI2GrBgiyI2vr9-pBtg1uRFMIN6Il-7Zatt940Og4NPPyF4zzt4LeE34Yb2Dg/https%3A%2F%2Fgithub.com%2FBioconductor%2FAnVIL%2Fpull%2F90%23issuecomment-1771491160>, or unsubscribe<https://secure-web.cisco.com/1pFrS_1OAVMaO8VvECDTkzxGIblR4v581-8rZDZ0in7JX4ZLrJOBW7ozagctBGB6FIKeDHz8wKofXZ3ho_Xf-owDXYfz1Ff_O64pPEdckJ2DUzMKszoGf8wJ2RxSFP1Zojp8KrcwXuR2CaCu4uhvyuy4MJdye5rMjLjK_sE_M-Y-F8PN-OrQmd3-wWVPGqIXyAIUHUapGtyb3HwpjtUW7UK5Zg5xcgBD1_Xs_4RVIHcLDFpG0NvhlueKQNBixO1ksgIKe4_YYVWNy2KXqXTH-qf3UlcWe7fOMFOgQUQUtAgCjNzpDzA7wHnPTeANZpli6/https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAECDIFCCXHBKWAOYF54CIK3YAFVD3AVCNFSM6AAAAAA4YPML22VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZRGQ4TCMJWGA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>


This email message may contain legally privileged and/or confidential information.  If you are not the intended recipient(s), or the employee or agent responsible for the delivery of this message to the intended recipient(s), you are hereby notified that any disclosure, copying, distribution, or use of this email message is prohibited.  If you have received this message in error, please notify the sender immediately by e-mail and delete this email message from your computer. Thank you.

**Review by LiNk-NY — APPROVED** [review]

**Pull request merged**